pkg/dns/ibm/common.go

```diff
@@ -0,0 +1,51 @@
+package common
+
+import (
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	iov1 "github.com/openshift/api/operatoringress/v1"
+)
+
+var (
+	// IBMResourceCRNRegexp is a regular expression of IBM resource CRNs
+	// See: https://cloud.ibm.com/docs/account?topic=account-crn
+	IBMResourceCRNRegexp = `^crn:v[0-9]:(?P<cloudName>[^:]*):(?P<cloudType>[^:]*):(?P<serviceName>[^:]*):(?P<location>[^:]*):(?P<scope>[^:]*):(?P<guid>[^:]*):(?P<resourceType>[^:]*):(?P<resourceID>[^:]*)$`
```

Comment:
This can be a const:
```suggestion
const (
	// IBMResourceCRNRegexp is a regular expression of IBM resource CRNs
	// See: https://cloud.ibm.com/docs/account?topic=account-crn
	IBMResourceCRNRegexp = `^crn:v[0-9]:(?P<cloudName>[^:]*):(?P<cloudType>[^:]*):(?P<serviceName>[^:]*):(?P<location>[^:]*):(?P<scope>[^:]*):(?P<guid>[^:]*):(?P<resourceType>[^:]*):(?P<resourceID>[^:]*)$`
```
Alternatively, you can define a compiled regexp (in which case it must be a var):
```suggestion
var (
	// IBMResourceCRNRegexp is a regular expression of IBM resource CRNs
	// See: https://cloud.ibm.com/docs/account?topic=account-crn
	IBMResourceCRNRegexp = regexp.MustCompile(`^crn:v[0-9]:(?P<cloudName>[^:]*):(?P<cloudType>[^:]*):(?P<serviceName>[^:]*):(?P<location>[^:]*):(?P<scope>[^:]*):(?P<guid>[^:]*):(?P<resourceType>[^:]*):(?P<resourceID>[^:]*)$`)
```

Reply:
Defined it as a compiled regexp var.